tiledb/ml/models/_base.py

```diff
@@ -94,6 +85,58 @@ def load(self, *, timestamp: Optional[Timestamp] = None) -> Model:
     def preview(self) -> str:
         """Abstract method that previews a machine learning model."""
 
+    def _get_file_properties(
+        self,
+    ) -> Mapping[str, str]:
+        if self.Framework != "TensorBoard":
+            return {
+                ModelFileProperties.TILEDB_ML_MODEL_ML_FRAMEWORK.value: self.Framework,
+                ModelFileProperties.TILEDB_ML_MODEL_ML_FRAMEWORK_VERSION.value: self.FrameworkVersion,
+                ModelFileProperties.TILEDB_ML_MODEL_STAGE.value: "STAGING",
+                ModelFileProperties.TILEDB_ML_MODEL_PYTHON_VERSION.value: platform.python_version(),
+                ModelFileProperties.TILEDB_ML_MODEL_PREVIEW.value: self.preview(),
+            }
+        else:
+            return {
+                TensorboardFileProperties.TILEDB_ML_TENSORBOARD_FRAMEWORK.value: self.Framework,
+                TensorboardFileProperties.TILEDB_ML_TENSORBOARD_VERSION.value: self.FrameworkVersion,
+            }
+
+    def _create_array(self, spec: Spec, **kwargs: Any) -> tiledb.Array:
+        """Internal method that creates a TileDB array based on the model's spec."""
+        dom = tiledb.Domain(
+            tiledb.Dim(
+                name=spec.domain_info[0],
+                domain=spec.domain_info[1],
+                tile=1,
+                dtype=np.int32,
+                ctx=self.ctx,
+            ),
+        )
+        attrs = [
+            tiledb.Attr(
+                name=field,
+                dtype=str if field == "layer_name" else bytes,
+                var=True,
+                filters=tiledb.FilterList([tiledb.ZstdFilter()])
+                if self.Framework != "TensorBoard"
```

Comment:
Same comment as in line 91: every `if self.Framework ... else ...` in the base class should be extracted as a method and overriden in subclasses if necessary.